corehq/apps/app_manager/decorators.py

```diff
@@ -73,7 +74,11 @@ def _safe_cached_download(request, *args, **kwargs):
             request.GET.pop('username')
 
         latest_enabled_build = None
-        if latest and username and toggles.MANAGE_RELEASES_PER_LOCATION.enabled(domain):
+        if latest and toggles.MANAGE_RELEASES_PER_LOCATION.enabled(domain):
+            if not username:
+                content_response = dict(error="app.update.not.allowed.user.logged_out",
+                                        default_response="Please log in the app to check for update.")
+                return HttpResponse(status=406, content=json.dumps(content_response))
```

Comment:
we check for `latest` bool in this which is set only when someone is requesting for an `update` so the initial app installation should proceed without this check.

Reply:
Okay.  Still seems scary to me - I'd check with mobile unless you're confident that that's okay.

Reply:
Sure, i will test this

Reply:
tested that the bool `latest` is set only in case of checking for update.
I installed a version using application scan barcode successfully. Didn't test ccz because thats an offline install anyway.